cmd/docker-mcp/internal/gateway/handlers.go

```diff
@@ -2,16 +2,41 @@ package gateway
 
 import (
 	"context"
+	"fmt"
+	"os"
+	"time"
 
 	"github.com/modelcontextprotocol/go-sdk/mcp"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/metric"
 
 	"github.com/docker/mcp-gateway/cmd/docker-mcp/internal/catalog"
+	"github.com/docker/mcp-gateway/cmd/docker-mcp/internal/telemetry"
 )
 
 func getClientConfig(readOnlyHint *bool, ss *mcp.ServerSession, server *mcp.Server) *clientConfig {
 	return &clientConfig{readOnly: readOnlyHint, serverSession: ss, server: server}
 }
 
+// inferServerType determines the type of MCP server based on its configuration
+func inferServerType(serverConfig *catalog.ServerConfig) string {
+	// Check for SSE endpoint first (including remote URLs)
+	if serverConfig.Spec.SSEEndpoint != "" || serverConfig.Spec.Remote.URL != "" {
```

Comment:
We are trying to deprecate SSEEndpoint.  I think we have already effectively done it.
For Remote, the Spec has the transport type and it will gradually shift to be streamable more and more.

Reply:
I only found one yaml file that showed what appeared to be valid values for Spec.Remote.Transport: sse and http (assuming streaming). 

Are these right?